src/keypair.rs

```diff
@@ -58,8 +63,20 @@ impl<KG: KeGroup, S: SecretKey<KG>> KeyPair<KG, S> {
 
 impl<KG: KeGroup> KeyPair<KG> {
     /// Generating a random key pair given a cryptographic rng
-    pub(crate) fn generate_random<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
-        let sk = KG::random_sk(rng);
+    pub(crate) fn generate_random<CS: voprf::CipherSuite, R: RngCore + CryptoRng>(
+        rng: &mut R,
+    ) -> Self
+    where
+        <CS::Hash as OutputSizeUser>::OutputSize:
+            IsLess<U256> + IsLessOrEqual<<CS::Hash as BlockSizeUser>::BlockSize>,
+    {
+        let mut scalar_bytes = GenericArray::<_, <KG as KeGroup>::SkLen>::default();
+        rng.fill_bytes(&mut scalar_bytes);
+        let sk = KG::derive_auth_keypair::<CS>(
+            GenericArray::clone_from_slice(&scalar_bytes),
+            &STR_OPAQUE_DERIVE_AUTH_KEY_PAIR,
+        )
```

Comment:
I believe we can just remove the `info` parameter from `derive_auth_keypair` and hardcode it into the method.